modules/objdetect/src/haar.cpp

```diff
@@ -1472,13 +1521,27 @@ class HaarDetectObjects_ScaleCascade_Invoker : public ParallelLoopBody
                 int result = cvRunHaarClassifierCascade( cascade, cvPoint(x, y), 0 );
                 if( result > 0 )
                 {
-                    mtx->lock();
-                    vec->push_back(Rect(x, y, winsize.width, winsize.height));
-                    mtx->unlock();
+                    vecLocal.push_back(Rect(x, y, winsize.width, winsize.height));
+
+                    if (vecLocal.size() >= PARALLEL_LOOP_BATCH_SIZE)
+                    {
+                        mtx->lock();
+                        vec->insert(vec->end(), vecLocal.begin(), vecLocal.end());
+                        mtx->unlock();
+
+                        vecLocal.clear();
+                    }
                 }
                 ixstep = result != 0 ? 1 : 2;
             }
         }
+
+        if (vecLocal.size())
```

Comment:
implicit conversion of size_t to bool is fine in C/C++, but it would be more stylish to use `!vecLocal.empty()` instead